nautobot_ssot/integrations/slurpit/diffsync/models/models.py

```diff
@@ -0,0 +1,330 @@
+        "role__name",
+        "serial",
+        "status__name",
+        "tags",
+        "system_of_record",
+        "last_synced_from_sor",
+    )
+    _children = {"inventory_item": "inventory_items"}
+
+    name: str
+    location__name: Optional[str] = None
+    location__location_type__name: Optional[str] = None
+    location__parent__name: Optional[str] = None
+    location__parent__location_type__name: Optional[str] = None
+    device_type__manufacturer__name: str
+    device_type__model: str
+    platform__name: Optional[str] = None
+    role__name: str
+    serial: Optional[str] = ""
+    status__name: str
+    inventory_items: List["InventoryItemModel"] = []
+    tags: List[TagDict] = []
+    system_of_record: Annotated[str, CustomFieldAnnotation(name="system_of_record", key="system_of_record")]
+    last_synced_from_sor: Annotated[str, CustomFieldAnnotation(name="last_synced_from_sor", key="last_synced_from_sor")]
+
+
+class InventoryItemModel(ModelQuerySetMixin, NautobotModel):
+    """Data model representing an InventoryItem."""
+
+    _model = InventoryItem
+    _modelname = "inventory_item"
+    _identifiers = ("name", "device__name")
+    _attributes = (
+        "description",
+        "part_id",
+        "serial",
+        "tags",
+        "system_of_record",
+        "last_synced_from_sor",
+    )
+
+    name: str
+    part_id: Optional[str]
+    serial: Optional[str]
+    description: Optional[str]
+    device__name: str
+    tags: List[TagDict] = []
+    system_of_record: Annotated[str, CustomFieldAnnotation(name="system_of_record", key="system_of_record")]
+    last_synced_from_sor: Annotated[str, CustomFieldAnnotation(name="last_synced_from_sor", key="last_synced_from_sor")]
+
+
+class VLANModel(ModelQuerySetMixin, NautobotModel):
+    """Data model representing a VLAN."""
+
+    _model = VLAN
+    _modelname = "vlan"
+    _identifiers = ("vid", "name")
+    _attributes = ("status__name", "tags", "system_of_record", "last_synced_from_sor")
+
+    vid: int
+    name: str
+    status__name: str
+    tags: List[TagDict] = []
+    system_of_record: Annotated[str, CustomFieldAnnotation(name="system_of_record", key="system_of_record")]
+    last_synced_from_sor: Annotated[str, CustomFieldAnnotation(name="last_synced_from_sor", key="last_synced_from_sor")]
+
+
+class VRFModel(ModelQuerySetMixin, NautobotModel):
+    """data model representing a VRF."""
+
+    _model = VRF
+    _modelname = "vrf"
+    _identifiers = ("name",)
+    _attributes = ("tags", "system_of_record", "last_synced_from_sor")
+
+    name: str
+    tags: List[TagDict] = []
+    system_of_record: Annotated[str, CustomFieldAnnotation(name="system_of_record", key="system_of_record")]
+    last_synced_from_sor: Annotated[str, CustomFieldAnnotation(name="last_synced_from_sor", key="last_synced_from_sor")]
+
+
+class VRFDict(TypedDict):
+    """TypedDict for VRF data."""
+
+    name: str
+
+
+class PrefixModel(ModelQuerySetMixin, NautobotModel):
```

Comment:
Any particular reason you didn't include Namespace? I could potentially see a use for allowing the user to specify the Namespace to import all IPAM data into so that differing environments don't potentially have an IP space collision. Maybe a future FR? Just thinking of potential issues I've seen for users regarding Prefixes and IP space.